inertia/middleware.go

```diff
@@ -0,0 +1,47 @@
+package inertia
+
+import (
+	"context"
+	"net/http"
+)
+
+type MiddlewareFunc func(http.Handler) http.Handler
+
+func Middleware(inertia *Inertia) MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ctx := context.WithValue(r.Context(), inertiaCtxKey, inertia)
+			req := r.WithContext(ctx)
```

Comment:
Should the intertia instance  still be added to the context if no header present?

Reply:
I think so, since for example the first non-Inertia request will not have the header present but will still need the inertia instance to render the full-html page response.